Asymmetric trainruns

louisgreiner opened the pull request:
# Description

Feature not perfect, but it's such a rabbit hole I could implement / fix features for a century.

> [!NOTE]
> About the consecutive times propagation: I keep the existing code identical since `travelTime.consecutiveTime` and `backwardTravelTime.consecutiveTime` are ***never*** called. However these might be nice be to refactored. (`propagateInitialConsecutiveTimes`, `propagateTrainrunInitialConsecutiveTimes`, `propagateConsecutiveTimesForTrainrun`, `propagateConsecutiveTimes`) => `TimeLockDto` is a bigger than the needs of the travel times

Ref https://github.com/OpenRailAssociation/netzgrafik-editor-frontend/issues/418

For reviewing: https://sncf.sharepoint.com/:p:/r/sites/OSRD644GrpO365/_layouts/15/Doc.aspx?sourcedoc=%7B68C024EF-2C05-4B4D-86C1-ED205F966F2F%7D&file=Cas%20modifs%20temps%20NGE.pptx&action=edit&mobileredirect=true

Missing documentation / video perhaps

# Issues

Some bugs (shaky features) to fix:
- symmetry recalculation on non-stop (only reset symmetry on first or last section currently)
- BackwardTravelTime text position on reticular bugged (inverted sometimes with the travelTime)

# Checklist

<!--
    Take care of the default items before marking your PR as ready for review,
    be prepared to add more items.
-->

- [x] This PR contains a description of the changes I'm making
- [x] I've read the [Contribution Guidelines](https://github.com/SchweizerischeBundesbahnen/netzgrafik-editor-frontend/blob/main/CONTRIBUTING.md)
- [ ] I've added tests for changes or features I've introduced
- [ ] I documented any high-level concepts I'm introducing in `documentation/`
- [x] CI is currently green and this is ready for review

<!--
    Please open PRs as Draft while you make CI green and/or finalise
    documentation.
-->

Commit:
asymmetry: introduce TrainrunSection asymmetric attributes

Getters, setters and util functions will be usefull later.

Fixed by the way an old wrong rebase on reconnect tests. Sorry for that.

Also fixed the associated tests, which updated other fields (like resources) that have been forgotten to be updated by 'fix: Resource deletion when node gets deleted'.

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: display asymmetry on reticular

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: introduce asymmetry arrow

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: enable filtering

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: adapt times services LeftAndRightTimeStructure updates

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: adapt times update functions to asymmetric (and non-stop) cases

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: introduce reset symmetry feature

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: introduce (re)set symmetry helper functions

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: helper functions for left/right <-> source/departure symmetry

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: trainrun-tab component

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: update trainrun-section-card component (one-way card)

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: trainrun-section-tab component

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

Commit:
asymmetry: perlenkette component

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

louisgreiner commented:
For the bug symmetry reset I edited:
> symmetry recalculation on non-stop (only reset symmetry on first or last section currently)
> ***SOLUTION : can we merge this as a new feature after the first merge?***

@aiAdrian @emersion what do you think? It will simplify the review

Commit:
asymmetry: fix travel times and name overlapping

We could also reduce the font size for travel times and name texts: $LINE_FONT_SIZE: 16px; -> $LINE_FONT_SIZE_REDUCED: 14px;

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

emersion commented on src/app/view/toggle-switch-button/toggle-switch-button.component.html:
```diff
@@ -1,7 +1,11 @@
+  <span
+    *ngIf="labelFalse"
+    [ngClass]="createLabelCheckedTag(false)"
+    (click)="!disabled && onToggle(false)"
```
Nit: would be a little bit less verbose to check for `disabled` in the `onToggle` function, instead of each time it's called.

Commit:
asymmetry: fix backward travel time displays

Adapted trainrun-section-tab and perlenkette for bottomTravelTime, and made backwardTravelTime displayed only if different of travelTime.

Signed-off-by: Louis Greiner <greiner.louis@gmail.com>

emersion reviewed (commented):
A few comments I've written when I've started reviewing below.

louisgreiner commented on src/app/view/toggle-switch-button/toggle-switch-button.component.html:
```diff
@@ -11,7 +15,11 @@
+    *ngIf="labelTrue"
+    [ngClass]="createLabelCheckedTag(true)"
+    (click)="!disabled && onToggle(true)"
+    [style.cursor]="disabled ? 'default' : 'pointer'"
```
I've done all of this is in the associated PR: https://github.com/OpenRailAssociation/netzgrafik-editor-frontend/pull/631/commits/bc09ecdaff44e460a5f9e35a66cb7e047a58efba

louisgreiner commented:
But you can test this branch anyway, the behavior should not be so different with @emersion 's implementation

Commit:
fixup! asymmetry: introduce asymmetry arrow

aiAdrian commented:
The feature is still in the latest version. This change was never polished and finished and since we still have the feature - we close this
